src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -77,8 +94,10 @@ public void StartFileChangeListeningAsync()
                 return newCookie;
             }, LazyThreadSafetyMode.ExecutionAndPublication);
 
-            // file change service is free-threaded. start running it in background right away
-            Task.Run(() => _fileChangeCookie.Value, CancellationToken.None);
+            lock (s_lastBackgroundTaskGate)
+            {
+                s_lastBackgroundTask = s_lastBackgroundTask.ContinueWith(_ => _fileChangeCookie.Value, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
For your second point the conversation has continued on the main conversation thread.

For your first point, looking at the docs:

    ExecuteSynchronously: ... If the antecedent is already complete when the continuation is created, the continuation will run on the thread that creates the continuation. ...

So yes, that very much reads our mutual concern is actually bad and we probably shouldn't try that.

Which lock was the one with the overhead? Any of our locks or the lock in the file change service?